pyphare/pyphare/pharesee/hierarchy.py

```diff
@@ -539,12 +575,103 @@ def __init__(self, patch_levels, domain_box, refinement_ratio=2, time=0., data_f
         if data_files is not None:
             self.data_files.update(data_files)
 
+        if len(self.quantities()) > 1:
+            for qty in self.quantities():
+                if qty in self.__dict__:
+                    continue
+                first = True
+                for time, levels in self.time_hier.items():
+                    new_lvls = {}
+                    for ilvl, level in levels.items():
+                        patches = []
+                        for patch in level.patches:
+                            patches += [Patch({qty: patch.patch_datas[qty]})]
+                        new_lvls[ilvl] = PatchLevel(ilvl, patches)
+                    if first:
+                        self.__dict__[qty] = PatchHierarchy(
+                            new_lvls, self.domain_box, time=time
+                        )
+                        first = False
+                    else:
+                        self.qty.time_hier[time] = new_lvls  # pylint: disable=E1101
+
+    def __getitem__(self, qty):
+        return self.__dict__[qty]
+
+    def __call__(self, qty=None, **kwargs):
+        def cuts(c, coord):
+            return c > coord.min() and c < coord.max()
+
+        class Extractor:
+            def __init__(self):
+                self.exclusions = []
+
+            def extract(self, coord, data):
+                mask = coord == coord
```

Comment:
I don't get the use of `coord == coord`, but it is certainly my ignorance...

Reply:
it makes an array of boolean of the same shape, set to True.